lib/src/main/java/org/team100/lib/barcode/Barcode.java

```diff
@@ -0,0 +1,85 @@
+package org.team100.lib.barcode;
+
+import java.util.function.IntSupplier;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.DigitalOutput;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/** Encapsulates the code reader. */
+public class Barcode implements IntSupplier {
+    private final DigitalOutput[] m_outputs;
+    private final Mux mux;
+    private final AnalogInput m_input;
+    private final DigitalOutput m_odd;
+    private final DigitalOutput m_even;
+    private final Sensor m_array;
+
+    private int id;
+
+    public Barcode() {
+        m_outputs = new DigitalOutput[] {
+                new DigitalOutput(0),
+                new DigitalOutput(1),
+                new DigitalOutput(2),
+                new DigitalOutput(3)
+        };
+        mux = new Mux(m_outputs);
+        m_input = new AnalogInput(0);
+        m_odd = new DigitalOutput(5);
+        m_even = new DigitalOutput(4);
+
+        // TODO: tune the thresholds
+        double[] thresholds = new double[] { 2.5, 3, 3, 2.4, 2.5 };
```

Comment:
is there something behind these numbers? is there some range of numbers that work, and other numbers that don't work?